pkg/git/github.go

```diff
@@ -290,6 +290,13 @@ func (g *GitHubGITProvider) GetPRThatCausedCommit(ctx context.Context, sha strin
 				return err
 			}
 			for _, pr := range closedPrs {
+				if pr == nil {
+					continue
+				}
```

Comment:
I'm surprised that `PullRequests.List` can return `nil`. I think it merits a comment.

Reply:
Have not seen it happen but because the sha could return nil I just wanted to add a check.